File: binarytodecimal.py
# Simple CLI Binary To Decimal & Vice Versa

def menu():
    print("Binary To Decimal & Decimal to Binary")
    print("1. Decimal To Binary")
    print("2. Binary To Decimal")
    print("3. Exit")

def decimalToBinary():
    try:
        decimal = int(input("Enter a decimal number:"))
        binary = bin(decimal).replace("0b", "")
        print(f"Converted Binary Value: {binary}")
    except ValueError:
        print("Please Enter a Valid Decimal")

def binaryToDecimal():
    try:
        conversion = int(input("Enter a binary number:"), 2)
        print(f"Converted Decimal Value: {conversion}")
    except ValueError:
        print("Please Enter a Valid Binary")

while 1:
    menu()
    try:
        choice = int(input("Enter your choice: "))
        if choice > 3:
            print("Please Enter a Valid Choice: ")
        if choice == 1:
            decimalToBinary()
        if choice == 2:
            binaryToDecimal()
        if choice == 3:
            break
    except ValueError:
        print("Please Enter a Valid Choice")

File: Golang/binarytodecimal.go
package golang

import (
	"fmt"
	"strconv"
)

func decimalToBinary() {
	var decimal int
	fmt.Print("Enter a decimal number: ") // No new line
	fmt.Scan(&decimal)                    // Takes input and puts it into the variable
	binary := strconv.FormatInt(int64(decimal), 2)
	// Conversion taken from https://www.techieindoor.com/go-program-to-convert-decimal-number-to-binary-in-go/
	// FromatInt takes only 64 bit integers and require conversion
	fmt.Printf("Converted Binary Value: %s\n", binary) // F string

}

func binaryToDecimal() {
	// Conversion taken from https://stackoverflow.com/questions/9271469/go-convert-string-which-represent-binary-number-into-int
	var binary string // based on the example, we take a string that represents the Binary
	fmt.Print("Enter a binary number: ")
	fmt.Scan(&binary)
	conversion, err := strconv.ParseInt(binary, 2, 64) // function returns both err and converted item
	if err != nil {                                    // nil is default response if successful
		fmt.Println("Please Enter a Valid Binary")
		return
	}
	fmt.Printf("Converted Decimal Value: %d\n", conversion)
}
